File: node/features/steps/circle.js
'use strict';

const should = require('should');
const assert = require('assert');
const { Given, When, Then } = require('cucumber');
const { radiusArea } = require('../../src/circle-area');
const { NegativeRadius } = require('../../src/exceptions/negative-radius');

let globalRadius = 0;
Given('un radio de valor {float}', function (radius) {
  globalRadius = radius;
  assert(!isNaN(radius));
});

When('al calcular el area', function () {
  assert(true);
  return;
});

When('al calcular el area, lanza una exception de tipo NegativeRadius', function () {
  try {
    const area = radiusArea(globalRadius);
    assert(false);
  } catch (NegativeRadius) {
    assert(true);
  }
});
 
Then('el area es igual a ~{float} con un error absoluto menor que {float}', function (value, absoluteErrorLine) {
  const area = radiusArea(globalRadius);
  const absoluteError = Math.abs(area - value);
  assert(absoluteError < absoluteErrorLine);
});

File: python/features/steps/circle.py
from behave import *
from src.circleArea import radiusArea

globalRadius = 0

@given('un radio de valor {radius}')
def step_impl(context, radius):
    radius = float(radius)

    global globalRadius
    globalRadius = radius
    assert globalRadius > 0

@when('al calcular el area')
def step_impl(context):
    pass

@then('el area es igual a ~{value} con un error absoluto menor que {absoluteErrorLine}')
def step_impl(context, value, absoluteErrorLine):
    value = float(value)
    absoluteErrorLine = float(absoluteErrorLine)
    
    global globalRadius
    area = radiusArea(globalRadius)
    absoluteError = abs(area - value)
    assert absoluteError < absoluteErrorLine